datafusion/functions-array/src/lib.rs

```diff
@@ -98,54 +103,245 @@ pub mod expr_fn {
+        );
+        functions.insert(
+            String::from("flatten"),
+            Box::new(flatten::flatten_udf) as _,
+        );
+        functions.insert(
+            String::from("array_sort"),
+            Box::new(sort::array_sort_udf) as _,
+        );
+        functions.insert(
+            String::from("array_repeat"),
+            Box::new(repeat::array_repeat_udf) as _,
+        );
+        functions.insert(
+            String::from("array_resize"),
+            Box::new(resize::array_resize_udf) as _,
+        );
+        functions.insert(
+            String::from("array_reverse"),
+            Box::new(reverse::array_reverse_udf) as _,
+        );
+        functions.insert(
+            String::from("array_distinct"),
+            Box::new(set_ops::array_distinct_udf) as _,
+        );
+        functions.insert(
+            String::from("array_intersect"),
+            Box::new(set_ops::array_intersect_udf) as _,
+        );
+        functions.insert(
+            String::from("array_union"),
+            Box::new(set_ops::array_union_udf) as _,
+        );
+        functions.insert(
+            String::from("array_position"),
+            Box::new(position::array_position_udf) as _,
+        );
+        functions.insert(
+            String::from("array_positions"),
+            Box::new(position::array_positions_udf) as _,
+        );
+        functions.insert(
+            String::from("array_remove"),
+            Box::new(remove::array_remove_udf) as _,
+        );
+        functions.insert(
+            String::from("array_remove_all"),
+            Box::new(remove::array_remove_all_udf) as _,
+        );
+        functions.insert(
+            String::from("array_remove_n"),
+            Box::new(remove::array_remove_n_udf) as _,
+        );
+        functions.insert(
+            String::from("array_replace"),
+            Box::new(replace::array_replace_udf) as _,
+        );
+        functions.insert(
+            String::from("array_replace_all"),
+            Box::new(replace::array_replace_all_udf) as _,
+        );
+        functions.insert(
+            String::from("array_replace_n"),
+            Box::new(replace::array_replace_n_udf) as _,
+        );
+
+
+
+        // TODO: Add more builtin functions here
+        Mutex::new(functions)
+    })
+}
+
+// Get an UDF by name
+//
+// Replace with `get_udf`
+// fn get_function_meta(&self, name: &str) -> Option<Arc<ScalarUDF>> {
+//     self.state.scalar_functions().get(name).cloned()
+// }
+pub fn get_array_udf(name: &str) -> Option<Arc<ScalarUDF>> {
+    array_functions().lock().unwrap().get(name).map(|f| f())
+}
+
+/// Register a single new UDF, so the user can register their own functions
+/// 
+/// Repalce old regsiter_udf
+pub fn register_array_udf(name: &str, udf: ScalarFactory) -> Option<ScalarFactory> {
+    // TODO: Check overwrite?
+    array_functions().lock().unwrap().insert(name.to_string(), udf)
```

Comment:
If we switch to static map for storing boxed fn impl, the user can only register with this method